### PR TITLE
Add automatic linting via golangci-lint

### PR DESCRIPTION
**Pull request by twelho**
This adds linting support via [golangci-lint](https://github.com/golangci/golangci-lint) to the CI pipeline on GitHub Actions. Also fixes all linting errors currently present in the codebase.

### CODE REVIEW

File: pkg/version/version.go [hunk 1]
```diff
@@ -22,7 +27,22 @@ type Info struct {
...
 
-// String returns info as a human-friendly version string.
-func (info Info) String() string {
+// String returns info as a human-friendly version string
+func (info *Info) String() string {
 	return info.GitVersion
 }
+
+// Get returns the version information of libgitops
+func Get() *Info {
```
Comment (luxas) [inline pkg/version/version.go:36]: Why did you add this?
Actually, I'm not sure we should have this package at all under `pkg/` really, it kind of belongs under the `cmd/` example only. What do you think?

File: pkg/storage/cache/index.go [hunk 2]
```diff
@@ -31,9 +31,9 @@ func (i *index) loadByID(gvk schema.GroupVersionKind, uid runtime.UID) (runtime.
 	return nil, nil
 }
 
-func (i *index) load(obj runtime.Object) (runtime.Object, error) {
-	return i.loadByID(obj.GroupVersionKind(), obj.GetUID())
-}
+//func (i *index) load(obj runtime.Object) (runtime.Object, error) {
```
Comment (luxas) [inline pkg/storage/cache/index.go:34]: maybe just remove?

File: pkg/version/version.go (hunk 1 above)
Comment (twelho) [reply pkg/version/version.go:36]: This was copied over from Ignite's behavior to get the linter to pass (make use of all the vars set by ldflags). I agree it doesn't really make sense for a library, so I'll move it over to the example.
